### PR TITLE
Refactor emscripten py pt3 - emscript_wasm_backend

### PR DESCRIPTION
**Pull request by jgravelle-google**
Final refactoring pass for now - did a bunch of function extraction in `emscript_wasm_backend`. Reused a few pieces from previous refactorings (yay!). Built this on top of part 2 (https://github.com/kripken/emscripten/pull/5150), but surprisingly to me the reuse I did happened to already be in part 1, so this is committable separately.

Tested with `default`, `asm2`, `binaryen0` and `binaryen2` with both backends. Skipped `other` and `browser` because these changes are pretty well localized to the `emscript_wasm_backend` path.

### CODE REVIEW

File: emscripten.py [hunk 1]
```diff
@@ -312,19 +310,26 @@ def update_settings_glue(settings, metadata):
   # Integrate info from backend
   if settings['SIDE_MODULE']:
```
Comment (jgravelle-google) [inline emscripten.py]: Thereby raising the question, should we include these settings changes in the wasm_backend path?

File: emscripten.py [hunk 2]
```diff
@@ -1762,18 +1625,96 @@ def save_settings():
...
+  with temp_files.get_file('.wb.s') as temp_s:
+    backend_args = create_backend_args_wasm(infile, temp_s, settings)
+    if DEBUG:
+      logging.debug('emscript: llvm wasm backend: ' + ' '.join(backend_args))
+      t = time.time()
+    shared.check_call(backend_args)
+    if DEBUG:
+      logging.debug('  emscript: llvm wasm backend took %s seconds' % (time.time() - t))
+      t = time.time()
+      import shutil
```
Comment (dschuff) [inline emscripten.py]: can we move these imports up? here and elsewhere.

File: emscripten.py (hunk 1 above)
Comment (dschuff) [inline emscripten.py]: you mean, should we implement SIDE_MODULE mode for the wasm backend?

Comment (jgravelle-google) [inline emscripten.py]: Forgot that it wasn't implemented, for one.

Mostly just, should we consolidate the setting of these settings. I see our options as:
1. leave this split into two functions, one of which supersets the other
2. combine these, but put the non-wasm-backend portions behind `if not is_wasm_backend():`, adding complexity to how the function executes
3. combine these, do possibly-irrelevant flag setting in the wasm-backend case, hope that the error is caught elsewhere

I guess what I'm saying is, 3 sounds least-bad to me, but it's a fairly weak preference, so I'm curious if anyone else has better ideas.

File: emscripten.py (hunk 2 above)
Comment (jgravelle-google) [inline emscripten.py]: sgtm https://github.com/kripken/emscripten/pull/5160/commits/e207842d4c04417f1cc39e1b8f15c3b84c775763

File: emscripten.py (hunk 1 above)
Comment (kripken) [inline emscripten.py]: 3 seems ok to me too.

Comment (jgravelle-google) [inline emscripten.py]: Cool, went with 3 in https://github.com/kripken/emscripten/pull/5160/commits/438b29117ce6259919216052be3c3c25abd8dcf1

**Comment by jgravelle-google** [comment]
Anything else to do here or is this good as-is?

**Comment by kripken** [comment]
Nothing looks bad to me, and nice refactoring, so lgtm assuming tests pass.

**Pull request merged**